server/src/client/Calendar.js

```diff
@@ -50,116 +55,146 @@ class App extends React.Component {
         rowSpan={cell.periodLength}
         colSpan={colSpan}
         key={key}
-        onClick={() => this.props.dispatch({
-          type: "TOGGLE_INFORMATION_DRAWER",
-          data: {
-            number: number,
-            information: cell
-          }
-        })}
-        style={{ backgroundColor: cell.color, cursor: "pointer" }}>
+        onClick={() =>
+          this.props.dispatch({
+            type: "TOGGLE_INFORMATION_DRAWER",
+            data: {
+              number: number,
+              information: cell,
+            },
+          })}
+        style={{ backgroundColor: cell.color, cursor: "pointer" }}
+      >
         <Typography variant="h7" style={{ color: "white" }}>
           {cell.name}
         </Typography>
       </TableCell>
-    )
+    );
   }
   generatePeriods(props) {
     //Generate everything
-    let calendarData = this.props.calendar;//needs to be changed
+    let calendarData = this.props.calendar; //needs to be changed
+    const semesterTerm = this.props.semesterVal;
     const rows = periods.map((period, periodKey) => {
-      if (periodTimesSummer["semesters"].includes(this.props.semesterVal) && periodTimesSummer[period] || periodTimesNormal["semesters"].includes(this.props.semesterVal) && periodTimesNormal[period]) {
+      if (
+        (semesterTerm.charAt(3) === "5" && periodTimesSummer[period]) ||
+        (semesterTerm.charAt(3) !== "5" && periodTimesNormal[period])
```

Comment:
is there a way to keep this logic on the BE? it doesn't really matter, but its cleaner to keep it in config

Reply:
done. all changes are in services/UF.js now.